tools/swift-stdlib-tool/swift-stdlib-tool.cpp

```diff
@@ -670,583 +642,568 @@ -(NSString *)sst_command {
-
-                // Work around authentication UI problems
-                // by signing one synchronously and then signing the rest.
-                [signingLock lock];
-                if (signedOne) {
-                    // First signer is complete. Proceed concurrently.
-                    [signingLock unlock];
-                } else {
-                    // We are the first signer. Hold the lock until we finish.
-                }
 
-                NSString *lib = key;
-                NSString *dst = [dst_dir stringByAppendingPathComponent:lib];
-
-                // Get the code signature, and copy the dylib to the side
-                // to preserve it in case it does not change.  We can use
-                // this to avoid unnecessary copies during delta installs
-                // to devices.
-                NSData *oldSignatureData = query_code_signature(dst);
-                const char *tmpFilePath = 0;
-                if (oldSignatureData) {
-                    // Make a copy of the existing file, with permissions and
-                    // mtime preserved.
-                    NSString *tmpFile = [dst stringByAppendingPathExtension:@"original"];
-                    tmpFilePath = tmpFile.fileSystemRepresentation;
-                    xcrunToolCommand(@[@"cp", @"-p", dst, tmpFile]);
-                }
+            // Proceed with (re-)codesigning.
+            log_v("Codesigning %s at %s", lib.c_str(), dst_dir.c_str());
 
-                // Proceed with (re-)codesigning.
-                log_v("Codesigning %s at %s", lib.fileSystemRepresentation, dst_dir.fileSystemRepresentation);
+            // Build the codesign invocation.
+            std::vector<std::string> commandAndArguments {
+                 "codesign", "--force", "--sign", ident, "--verbose"};
 
-                // Build the codesign invocation.
-                NSMutableArray *commandAndArguments = 
-                    [NSMutableArray arrayWithObjects:
-                     @"codesign", 
-                     @"--force", @"--sign", ident, @"--verbose", nil];
-
-                if (keychain) {
-                    [commandAndArguments addObject:@"--keychain"];
-                    [commandAndArguments addObject:keychain];
-                }
+            if (!keychain.empty()) {
+                commandAndArguments.push_back("--keychain");
+                commandAndArguments.push_back(keychain);
+            }
 
-                // Other codesign flags come later
-                // so they can override the default flags.
-                [commandAndArguments addObjectsFromArray:otherCodesignFlags];
+            // Other codesign flags come later
+            // so they can override the default flags.
+            std::move(otherCodesignFlags.begin(), otherCodesignFlags.end(),
+                      std::back_inserter(commandAndArguments));
```

Comment:
I am not sure that moving in every loop from `otherCodesignFlags` is the correct thing. Why not `std::copy` which should be clearly safe?